compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -213,7 +213,10 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
           }
         case tree: TypeApply =>
           val tree1 @ TypeApply(fn, args) = normalizeTypeArgs(tree)
-          Checking.checkBounds(args, fn.tpe.widen.asInstanceOf[PolyType])
+          if (fn.symbol != defn.ChildAnnot.primaryConstructor) {
```

Comment:
Instead of special-casing Child, the compiler could be changed to always use it with fully-applied types with wildcards, e.g. `Child[Cons[_]]`